forte/data/ontology/core.py

```diff
@@ -429,7 +432,20 @@ def _resolve_pointer(self, ptr: BasePointer) -> Entry:
         if isinstance(ptr, Pointer):
             return self.pack.get_entry(ptr.tid)
         elif isinstance(ptr, MpPointer):
-            return self.pack.packs[ptr.pack_index].get_entry(ptr.tid)
+            # bugfix/new feature 559: in new version pack_index will be using pack_id internally
+            pack_array_index = ptr.pack_index  # old version
+            pack_version = [0, 0, 0]
+            try:
+                pack_version = self.pack.pack_version
+            except AttributeError as ae:
+                print(ae)  # the version is not set : print or log the exception
+
+            if pack_version >= BACKWARD_COMPATIBLE_VER:
```

Comment:
Version handling could be more complex, how about we still use string versions (just like the python or library version), and use version parsing tools? Maybe this one (https://pypi.org/project/version-parser/), but if there are better alternatives it'd be great too.

Reply:
Or something along the line with this: https://packaging.pypa.io/en/latest/version.html